win32ss/user/user32/windows/text.c

```diff
@@ -148,13 +148,24 @@ LPSTR
 WINAPI
 CharPrevA(LPCSTR start, LPCSTR ptr)
 {
-    while (*start && (start < ptr))
+    if (ptr > start)
     {
-        LPCSTR next = CharNextA(start);
-        if (next >= ptr) break;
-        start = next;
+        --ptr;
+        if (gpsi->dwSRVIFlags & SRVINFO_DBCSENABLED)
```

Comment:
So it's not possible to recycle CharNext(Ex)A here anymore?

Reply:
Yes, this is what makes it a bit different.

Reply:
If there is a codepage without lead bytes you might specify that incase `SRVINFO_DBCSENABLED` is FALSE,
and then you can call the `CharPrevExA` from `CharPrevA`

Reply:
I have tried wrapping `CharPrevA` around `CharPrevExA(CP_ACP, ...)`, seems to work fine, and there are no additional failed tests by doing so. However this way it's consistent with `CharNext(Ex)A` implementation.